src/itkOMEZarrNGFFImageIO.cxx

```diff
@@ -190,6 +190,33 @@ ReadFromStore(const tensorstore::TensorStore<> & store, const ImageIORegion & st
   }
 }
 
+// Reads from the store if the specified pixel type and the ITK component type match.
+template <typename TPixel>
+bool
+ReadFromStoreIfTypesMatch(const IOComponentEnum              componentType,
+                          const tensorstore::TensorStore<> & store,
+                          const ImageIORegion &              storeIORegion,
+                          void *                             buffer)
+{
+  if (tensorstoreToITKComponentType(tensorstore::dtype_v<TPixel>) == componentType)
+  {
+    ReadFromStore(store, storeIORegion, static_cast<TPixel *>(buffer));
+    return true;
+  }
+  return false;
+}
+
+// Tries to read from the specified store, trying any of the specified pixel types.
+template <typename... TPixel>
+bool
+TryToReadFromStore(const IOComponentEnum              componentType,
+                   const tensorstore::TensorStore<> & store,
+                   const ImageIORegion &              storeIORegion,
+                   void *                             buffer)
+{
+  return (ReadFromStoreIfTypesMatch<TPixel>(componentType, store, storeIORegion, buffer) || ...);
```

Comment:
Thanks for merging, Matt! The most interesting part of this commit, IMhO, is the [fold expression](https://en.cppreference.com/w/cpp/language/fold) above here, which expands to something like (in pseudo code):

```cpp
return Read<int8_t>() || Read<uint8_t>() || ... || Read<float>() || Read<double>();
```